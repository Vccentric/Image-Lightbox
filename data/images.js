import img1 from "../public/images/img_mountains_wide.jpg";
import img2 from "../public/images/img_nature_wide.jpg";
import img3 from "../public/images/img_snow_wide.jpg";

export const images = [
  { src: img1, alt: "Spring Mountains" },
  { src: img2, alt: "Fall Mountains" },
  { src: img3, alt: "Winter Mountains" },
];

export default images;
