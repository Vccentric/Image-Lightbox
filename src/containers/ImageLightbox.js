import React, { useState } from "react";
import PropTypes from "prop-types";
import Lightbox from "../components/Lightbox";

// initial state
const initState = {
  toggleOpen: false,
  currentIndex: 0,
};

// main container component
const ImageLightbox = ({ data = {}, localState = initState }) => {
  // setup state
  const [lightboxState, setLightboxState] = useState(localState);
  const { toggleOpen, currentIndex } = lightboxState;
  const max = data.images?.length || 0;

  // setup functions
  const close = () => {
    setLightboxState({
      toggleOpen: false,
      currentIndex,
    });
  };
  const prev = () => {
    let index = currentIndex - 1;
    index = index < 0 ? max - 1 : index;
    setLightboxState({
      toggleOpen,
      currentIndex: index,
    });
  };
  const next = () => {
    let index = currentIndex + 1;
    index = index >= max ? 0 : index;
    setLightboxState({
      toggleOpen,
      currentIndex: index,
    });
  };

  // setup elements
  const imageItems = data.images?.map((image, index) => {
    // setup variables
    const key = index * Math.random();

    // setup functions
    const selectedImage = () => {
      setLightboxState({
        toggleOpen: true,
        currentIndex: index,
      });
    };

    return (
      <li key={key}>
        <img
          className="thumbnail"
          onClick={selectedImage}
          src={image.src}
          alt={image.alt}
        />
      </li>
    );
  });

  return (
    <div id="image-lightbox">
      <h1 id="header">Image Lightbox</h1>
      <ul id="image-grid">{imageItems}</ul>
      <Lightbox
        toggleOpen={toggleOpen}
        currentImage={data?.images?.[currentIndex]}
        close={close}
        prev={prev}
        next={next}
      />
    </div>
  );
};

ImageLightbox.propTypes = {
  data: PropTypes.object,
  localState: PropTypes.object,
};

export default ImageLightbox;
