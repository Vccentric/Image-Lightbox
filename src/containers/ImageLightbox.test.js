import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import React from "react";
import ImageLightbox from "./ImageLightbox";
import images from "../../data/images";

// setup variables
const data = { images };
const initState = {
  toggleOpen: false,
  currentIndex: 0,
};

test("renders correctly - snapshot", () => {
  const tree = renderer
    .create(<ImageLightbox data={data} localState={initState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders image lightbox with lightbox toggle closed", () => {
  const { container } = render(
    <ImageLightbox data={data} localState={initState} />
  );
  expect(container.querySelector("#lightbox")?.className).toEqual("close");
});

test("renders image lightbox with lightbox toggle open", () => {
  const mockState = initState;
  mockState.toggleOpen = true;
  const { container } = render(
    <ImageLightbox data={data} localState={mockState} />
  );
  expect(container.querySelector("#lightbox")?.className).toEqual("");
});
