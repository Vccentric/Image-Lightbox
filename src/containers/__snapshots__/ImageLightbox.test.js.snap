// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - snapshot 1`] = `
<div
  id="image-lightbox"
>
  <h1
    className="header"
  >
    Image Lightbox
  </h1>
  <ul
    id="image-grid"
  >
    <li>
      <img
        alt="Spring Mountains"
        className="thumbnail"
        onClick={[Function]}
        src=""
      />
    </li>
    <li>
      <img
        alt="Fall Mountains"
        className="thumbnail"
        onClick={[Function]}
        src=""
      />
    </li>
    <li>
      <img
        alt="Winter Mountains"
        className="thumbnail"
        onClick={[Function]}
        src=""
      />
    </li>
  </ul>
  <div
    className="close"
    id="lightbox"
  >
    <div
      id="overlay"
    />
    <div
      id="modal"
    >
      <button
        className="close"
        onClick={[Function]}
      >
        X
      </button>
      <div
        className="image-container"
      >
        <div
          className="full-image"
        >
          <img
            alt="Spring Mountains"
            className="selected-image"
            onClick={[Function]}
            src=""
          />
        </div>
        <span
          className="caption"
        >
          Spring Mountains
        </span>
      </div>
      <div
        className="controls-container"
      >
        <button
          className="prev"
          onClick={[Function]}
        >
          &lt;&lt; PREV
        </button>
        <button
          className="next"
          onClick={[Function]}
        >
          NEXT &gt;&gt;
        </button>
      </div>
    </div>
  </div>
</div>
`;
